lib/providers/autoCompleteHelper.js

```diff
@@ -182,208 +180,42 @@ export default {
-
-		const appcPath = path.join(homedir(), '.appcelerator/install');
-		const version = fs.readFileSync(path.join(appcPath, '.version'), 'utf8');
-		const alloyPath = path.join(appcPath, version, 'package/node_modules/alloy');
-		// const alloyApi = JSON.parse(fs.readFileSync(path.join(alloyPath, 'docs/api.jsca'), 'utf8'));
-		const sdk = Appc.selectedSdk();
-		const titaniumAPIPath = path.join(sdk.path, 'api.jsca');
-		const api = JSON.parse(fs.readFileSync(titaniumAPIPath, 'utf8'));
-
-		// generate tag list
-		const fns = fs.readdirSync(path.join(alloyPath, '/Alloy/commands/compile/parsers'));
-		let tagDic = {};
-		for (const fn of fns) {
-			if (!fn.endsWith('.js')) {
-				continue;
-			}
-			const ar = fn.split('.');
-			const tagName = ar[ar.length - 2];
-			if (tagName.indexOf('_') !== 0 && tagName[0] === tagName[0].toUpperCase()) {
-				tagDic[tagName] = {
-					apiName: fn.replace('.js', '')
-				};
-			} else if (tagName === '_ProxyProperty' && fn.indexOf('Ti.UI') === 0) {
-				tagDic[ar[ar.length - 3]] = { // Ti.UI.Window._ProxyProperty
-					apiName: fn.replace('.js', '').replace('._ProxyProperty', '')
-				};
-			}
-		}
-
-		// add missing tags
-		Object.assign(tagDic, {
-			View: {
-				apiName: 'Ti.UI.View'
-			},
-			Templates: {},
-			HeaderView: {},
-			FooterView: {},
-			ScrollView: {
-				apiName: 'Ti.UI.ScrollView'
-			},
-			Slider: {
-				apiName: 'Ti.UI.Slider'
-			},
-			TableViewRow: {
-				apiName: 'Ti.UI.TableViewRow'
-			},
-			Alloy: {},
-			ActivityIndicator: {
-				apiName: 'Ti.UI.ActivityIndicator'
-			},
-			WebView: {
-				apiName: 'Ti.UI.WebView'
-			}
-		});
-
-		// property list
-		const types = {};
-		const props = {};
-		api.types.forEach((type) => {
-			if (type.deprecated) {
-				return;
-			}
-
-			let propertyNamesOfType = [];
-			type.properties.forEach((prop) => {
-				if (prop.permission !== 'read-only' && prop.name.indexOf('Modules.') !== 0) {
-
-					propertyNamesOfType.push(prop.name);
-
-					if (prop.deprecated) {
-						prop.name += '|deprecated';
-					}
-
-					// property name
-					if (props[prop.name]) { // if duplicated property name - merge available values
-						Object.assign(props[prop.name], {
-							description: props[prop.name].description === prop.description.replace(/<p>|<\/p>/g, '') ? props[prop.name].description : ''
-						});
-						if (prop.constants.length) {
-							const values = props[prop.name].values ? props[prop.name].values.concat(prop.constants) : prop.constants;
-							props[prop.name].values = [ ...new Set(values) ];
-						}
-
-					} else {
-						props[prop.name] = {
-							description: prop.description.replace(/<p>|<\/p>/g, ''),
-							type: prop.type
-						};
-						if (prop.constants.length) {
-							props[prop.name].values = prop.constants;
-						}
-					}
+	async generateAutoCompleteSuggestions() {
+		const force = atom.config.get('appcelerator-titanium.general.generateAutoCompleteSuggestions');
```

Comment:
I'm not sure whether I'd like to keep this config setting. I think we could add something like the vscode extension where we have a command (or a menu item maybe?) that does this on the fly rather then requiring a user to restart their editor